protos/graph.proto

```diff
@@ -0,0 +1,208 @@
+//
+// Bentobox
+// Proto
+// Graph
+//
+
+syntax = "proto3";
+package bento;
+
+import "values.proto";
+import "references.proto";
+
+// Node in a executable graph that represents a computable/evaluatable operation.
+message Node {
+    // Constant - node that always outputs its held value
+    message Const {
+        // Value held by this constant
+        Value held_value = 1;
+    }
+
+    /* Component access, mutation */
+    // Retrieve - node that retrieves a value from a entity's component
+    message Retrieve {
+        // Reference to the attribute to retriev
+        AttributeRef retrieve_attr = 1;
+    }
+
+    // Mutate - node mutate the value in an entity's component based 
+    // on value given by a node
+    message Mutate {
```

Comment:
Just something to consider in the future, you could add in some idea of expressions vs statements. An expression would be something that returns a value while a statement does not. So, nodes such as `Const`, `Add` or even [`Switch`](https://kotlinlang.org/docs/reference/control-flow.html#if-expression) can be expressions while `Mutate` is a statement.

What I mean is something like:
```message Add {
    Node.Expression x = 1;
    Node.Expression y = 1;
}
```

Reply:
Interesting idea, but right now we want to keep the data model as simple as possible, so I want to avoid introducing new concepts.